insights/parsers/nftables.py

```diff
@@ -0,0 +1,154 @@
+"""
+Nftables Configurations
+=======================
+
+This module includes the following parsers:
+
+NftListRuleSet - command ``nft -j list ruleset``
+------------------------------------------------
+"""
+
+from collections import defaultdict
+
+from insights.core import JSONParser
+from insights.core.dr import SkipComponent
+from insights.core.plugins import parser
+from insights.specs import Specs
+
+
+@parser(Specs.nft_list_ruleset)
+class NftListRuleSet(JSONParser):
+    """
+    It parses the output of "nft -j list ruleset".
+
+    Sample output::
+
+        {
+            "nftables": [
+                {
+                    "metainfo": {
+                        "version": "0.9.3", "release_name": "Topsy", "json_schema_version": 1
+                    }
+                },
+                {
+                    "table": {
+                        "family": "ip", "name": "example_table1", "handle": 17
+                    }
+                },
+                {
+                    "map": {
+                        "family": "ip", "name": "example_map", "table": "example_table1", "type": "ipv4_addr", "handle": 2, "map": "verdict", "elem": [["192.0.2.1", {"accept": null}], ["192.0.2.2", {"drop": null}], ["192.0.2.3", {"accept": null}]]
+                    }
+                },
+                {
+                    "set": {
+                        "family": "ip", "name": "example_set", "table": "example_table1", "type": "ipv4_addr", "handle": 4
+                    }
+                },
+                {
+                    "chain": {
+                        "family": "ip", "table": "example_table1", "name": "example_chain1", "handle": 1, "type": "filter", "hook": "input", "prio": 0, "policy": "accept"
+                    }
+                },
+                {
+                    "rule": {
+                        "family": "ip", "table": "example_table1", "chain": "example_chain1", "handle": 3, "expr": [{"vmap": {"key": {"payload": {"protocol": "ip", "field": "saddr"}}, "data": "@example_map"}}]
+                    }
+                },
+                {
+                    "rule": {
+                        "family": "ip", "table": "example_table1", "chain": "example_chain1", "handle": 7, "expr": [{"match": {"op": "==", "left": {"payload": {"protocol": "tcp", "field": "dport"}}, "right": 22}}, {"counter": {"packets": 29, "bytes": 1976}}, {"accept": null}]
+                    }
+                }
+            ]
+        }
```

Comment:
This sample looks like quite large inside a docstrings example.  I'm not sure but ... will it be possible to reduce/shorten some items accordingly ...